src/main/java/competition/subsystems/vision/VisionSubsystem.java

```diff
@@ -153,11 +160,20 @@ public Optional<EstimatedRobotPose> getRearPhotonVisionEstimatedPose(Pose2d prev
         }
     }
 
-    private boolean isEstimatedPoseReliable(EstimatedRobotPose estimatedPose) {
+    public boolean isEstimatedPoseReliable(EstimatedRobotPose estimatedPose) {
         if (estimatedPose.targetsUsed.size() == 0) {
             return false;
         }
 
+        // Pose isn't reliable if we see a tag id that shouldn't be on the field
+        var allTagIds = estimatedPose.targetsUsed.stream()
+                .map(target -> target.getFiducialId()).toList();
+        if (allTagIds.stream().anyMatch(id -> id < 1 || id > 8)) {
+            this.log.warn("Ignoring vision pose with invalid tag id. Visible tags: "
```

Comment:
this log could end up streaming a lot, once per cycle right? Maybe we can latch to only do it once before it gets better?

Reply:
Photon Vision was already logging any time it saw an invalid id, this just replaces that log (with a full list of tags, in case there is more than one)